### PR TITLE
TDR-3729 Push dependency update to a branch and create script for creating PR from branch

### PR DESCRIPTION
**Pull request by thanhz**
This pr does the following:

- Modifies the `repos.md` file adding the target branch (`scala-steward-dependencies`) from where it will create PR's to.
- Adds a new script that will create a PR from the `scala-steward-dependencies` branch for all repos in `repos.md` (This will be a crom job but just leaving that out for the moment so I can have more control)

If everything is okay in this PR. I will create the target branch manually for all the repositories and make a pr to add the required mergify conf and script that keeps the target branch up to date with master for each repository before I merge this pr.

### CODE REVIEW

File: .github/workflows/create-dependencies-pull-requests.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Read repositories from file
+        id: read-repos
+        run: |
+            repos_content=""
+            while IFS= read -r line; do
+            repo_name=$(echo "$line" | sed -e 's/^- //' -e 's/:steward-dependencies//')
```
Comment (TomJKing) [inline .github/workflows/create-dependencies-pull-requests.yml:18]: Maybe rename the branch: `scala-steward-dependencies`

File: .github/workflows/create-dependencies-pull-requests.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            repos_content+="$repo_name "
+            done < repos.md
+            echo -e "::set-output name=repos::$repos_content"
+        shell: bash
+
+      - name: Create Pull Request
+        run: |
+          for repo in ${{ steps.read-repos.outputs.repos }}; do
+            default_branch=$(gh api /repos/$repo -q '.default_branch')
+            develop=$(gh api /repos/$repo/commits/refs/heads/develop -q '.sha')
```
Comment (TomJKing) [inline .github/workflows/create-dependencies-pull-requests.yml:28]: Should the name of the "develop" branch not be "scala-steward-dependencies"? Cold be misunderstanding what is happening here.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of minor comments

File: .github/workflows/create-dependencies-pull-requests.yml (hunk 2 above)
Comment (thanhz) [reply .github/workflows/create-dependencies-pull-requests.yml]: Ah I totally missed this bit. I've corrected the branch name

**Review by TomJKing — COMMENTED** [review]
Sorry, just noticed one more small change needed on re-review.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**